.gitlab/kernel_version_testing/system_probe.yml

```
@@ -383,37 +388,45 @@ kernel_matrix_testing_run_tests_arm64:
     - .kernel_matrix_testing_run_tests
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_arm64$DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
   tags: ["arch:arm64"]
-  needs: ["kernel_matrix_testing_setup_env", "upload_dependencies_arm64", "upload_system_probe_tests_arm64", "upload_minimized_btfs_arm64"]
+  needs: ["kernel_matrix_testing_setup_env_arm64", "upload_dependencies_arm64", "upload_system_probe_tests_arm64", "upload_minimized_btfs_arm64"]
   timeout: 3h
   variables:
     ARCH: "arm64"
   parallel:
     matrix:
       - TAG: ["ubuntu_20.04", "ubuntu_22.04", "amzn_4.14", "amzn_5.4", "amzn_5.10", "fedora_37", "fedora_38"]
 
-kernel_matrix_testing_cleanup:
+.kernel_matrix_testing_cleanup:
   stage: kernel_matrix_testing
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions/runner$TEST_INFRA_DEFINITIONS_BUILDIMAGES_SUFFIX:$TEST_INFRA_DEFINITIONS_BUILDIMAGES
-  needs: ["kernel_matrix_testing_setup_env", "kernel_matrix_testing_run_tests_x64", "kernel_matrix_testing_run_tests_arm64"]
   when: always
   tags: ["arch:amd64"]
   rules:
     !reference [ .on_system_probe_changes_or_manual ]
   before_script:
     - !reference [.kernel_matrix_testing_new_profile]
   script:
-    - |
-      if [ -f $CI_PROJECT_DIR/stack.outputs ]; then
-        INSTANCE_IPS=$(cat $CI_PROJECT_DIR/stack.outputs | grep -e '-instance-ip' | cut -d ' ' -f 2 | tr '\n' ' ' | awk '{$1=$1};1' | tr ' ' ',')
-        IP_FILTER="Name=private-ip-address,Values=$INSTANCE_IPS"
-      else
-        echo "No ${CI_PROJECT_DIR}/stack.outputs file found"
-        IP_FILTER=""
-      fi
     - !reference [.shared_filters_and_queries]
-    - INSTANCE_IDS=$(aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_PIPELINE  $IP_FILTER --output text --query $QUERY_INSTANCE_IDS | tr '\n' ' ')
-    - echo $INSTANCE_IDS
+    - aws ec2 describe-instances --filters $FILTER_TEAM $FILTER_MANAGED $FILTER_PIPELINE $FILTER_ARCH $FILTER_INSTANCE_TYPE --output json --query $QUERY_INSTANCE_IDS | tee -a instance.json
+    - INSTANCE_ID="$(jq -r '.[0][0]' < instance.json)"
+    - echo ${INSTANCE_ID}
     - |
-      if [ $(echo $INSTANCE_IDS | wc -w) != "0" ]; then
-        aws ec2 terminate-instances --instance-ids $INSTANCE_IDS
+      if [[ "${INSTANCE_ID}" != "" ]]; then
```

It looks like `jq -r '.[0][0]` outputs null if the aws command returns an empty list `[]`
```suggestion
      if [[ "${INSTANCE_ID}" != "null" ]]; then
```